deep_qa/contrib/background_search/bow_lsh.py

```diff
@@ -0,0 +1,170 @@
+        # interval for UNK if needed.
+        self.vector_max = -float("inf")
+        self.vector_min = float("inf")
+        self.en_nlp = spacy.load('en')
+        self.indexed_background = {}  # index -> background sentence
+
+    def read_embeddings_file(self, embeddings_file: str):
+        with gzip.open(embeddings_file, 'rb') as embeddings_file:
+            for line in embeddings_file:
+                fields = line.decode('utf-8').strip().split(' ')
+                self.embedding_dim = len(fields) - 1
+                word = fields[0]
+                vector = numpy.asarray(fields[1:], dtype='float32')
+                vector_min = min(vector)
+                vector_max = max(vector)
+                if vector_min < self.vector_min:
+                    self.vector_min = vector_min
+                if vector_max > self.vector_max:
+                    self.vector_max = vector_max
+                self.embeddings[word] = vector
+
+    def load_model(self):
+        pickled_embeddings_file = "%s/embeddings.pkl" % self.serialization_prefix
+        pickled_lsh_file = "%s/lsh.pkl" % self.serialization_prefix
+        self.embeddings = pickle.load(open(pickled_embeddings_file, 'rb'))
+        indexed_background_file = open("%s/background.tsv" % self.serialization_prefix, "r")
+        for vector in self.embeddings.values():
+            if self.embedding_dim is None:
+                self.embedding_dim = len(vector)
+                vector_min = min(vector)
+                vector_max = max(vector)
+                if vector_min < self.vector_min:
+                    self.vector_min = vector_min
+                if vector_max > self.vector_max:
+                    self.vector_max = vector_max 
+        self.lsh = pickle.load(open(pickled_lsh_file, 'rb'))
+        for line in indexed_background_file:
+            parts = line.strip().split('\t')
+            self.indexed_background[int(parts[0])] = parts[1]
+
+    def save_model(self):
+        if not os.path.exists(self.serialization_prefix):
+            os.makedirs(self.serialization_prefix)
+        pickled_embeddings_file = open("%s/embeddings.pkl" % self.serialization_prefix, "wb")
+        pickled_lsh_file = open("%s/lsh.pkl" % self.serialization_prefix, "wb")
+        indexed_background_file = open("%s/background.tsv" % self.serialization_prefix, "w")
+        print("\tDumping embeddings", file=sys.stderr)
+        pickle.dump(self.embeddings, pickled_embeddings_file)
+        print("\tDumping sentences", file=sys.stderr)
+        for index, sentence in self.indexed_background.items():
+            sentence = sentence.replace('\t', ' ')  # Sanitizing sentences before making a tsv.
+            print("%d\t%s" % (index, sentence), file=indexed_background_file)
+        pickled_embeddings_file.close()
+        indexed_background_file.close()
+        # Serializing the LSH is memory intensive. Deleting the other members first.
```

Comment:
You should note in the docstring of this method that the object is unusable after saving.